src/main/java/org/jenkinsci/plugins/configfiles/ConfigFilesManagement.java

```diff
@@ -188,25 +184,40 @@ public void doEditConfig(StaplerRequest req, StaplerResponse rsp, @QueryParamete
      * @throws IOException
      * @throws ServletException
      */
-    public void doAddConfig(StaplerRequest req, StaplerResponse rsp, @QueryParameter("providerId") String providerId) throws IOException, ServletException {
+    public void doAddConfig(StaplerRequest req, StaplerResponse rsp, @QueryParameter("providerId") String providerId, @QueryParameter("configId") String configId) throws IOException, ServletException {
         checkPermission(Hudson.ADMINISTER);
 
-        for (ConfigProvider provider : ConfigProvider.all()) {
-            if (provider.getProviderId().equals(providerId)) {
-                req.setAttribute("contentType", provider.getContentType());
-                req.setAttribute("provider", provider);
-                Config config = provider.newConfig();
-                req.setAttribute("config", config);
-                break;
-            }
+        if (providerId == null || providerId.isEmpty()) {
+            // TODO add a warning message: you gorgot to select a provider
```

Comment:
> can you recommend me a way to add an error message if the user has not selected a provider?

[This](https://github.com/jenkinsci/jenkins/blob/6641190b8cce326bf9040de4d8182669ff7582d3/core/src/main/resources/lib/hudson/newFromList/form.jelly#L102-L119) is the implementation for the _New Item_ page.